forte/data/extractor/predictor.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=logging-fstring-interpolation
+import logging
+import os
+from abc import abstractmethod
+from typing import Dict, List, Optional, Tuple, Type, Any, Union, Iterable, Generic, Iterator
+
+import numpy as np
+import torch
+import itertools
+from copy import copy
+
+from forte.common.configuration import Config
+from forte.common.resources import Resources
+from forte.data.data_pack import DataPack
+from forte.data.ontology import Annotation
+from forte.data.ontology.core import Entry
+from forte.data.types import DataRequest
+from forte.models.ner import utils
+from forte.data.base_pack import PackType
+from forte.models.ner.model_factory import BiRecurrentConvCRF
+from forte.processors.base.base_processor import BaseProcessor
+from ft.onto.base_ontology import Token, Sentence, EntityMention
+from forte.data.extractor.extractor import TextExtractor, AnnotationSeqExtractor
+from forte.data.extractor.converter import Converter
+from forte.process_manager import ProcessJobStatus
+from forte.data import slice_batch
+from forte.data.data_utils_io import merge_batches, batch_instances
+
+
+class Predictor(BaseProcessor):
+    def __init__(self):
+        super().__init__()
+        self.pack_pools = []
+        self.instance_pools = []
+        self.features_pools = []
+
+    def initialize(self, resources: Resources, configs: Optional[Config]):
+        super().initialize(resources, configs)
+        # TODO: load these params from outside
+        self.batch_size = 11
+        self.scope = Sentence
+
+        config1 = {
+            "scope": Sentence,
+            "entry": Token,
+            "vocab_use_unk": True
+        }
+        extractor1 = TextExtractor(config1)
```

Comment:
If your goal is to create the `extractor` here, how about creating them outside and pas them in?

Reply:
Yes. Right now I am loading these from the config from trainpipeline.